mllib/src/main/scala/org/apache/spark/ml/stat/Summarizer.scala

```diff
@@ -527,27 +570,28 @@ private[ml] object SummaryBuilderImpl extends Logging {
       weightExpr: Expression,
       mutableAggBufferOffset: Int,
       inputAggBufferOffset: Int)
-    extends TypedImperativeAggregate[SummarizerBuffer] {
+    extends TypedImperativeAggregate[SummarizerBuffer] with ImplicitCastInputTypes {
 
-    override def eval(state: SummarizerBuffer): InternalRow = {
+    override def eval(state: SummarizerBuffer): Any = {
```

Comment:
why change the return type?

Reply:
Both of them works, but other similar aggregate function also use `Any`. Will it cause some issues ?